contribute-docs/style-guide/seo.md

```diff
@@ -0,0 +1,452 @@
+# SEO guidelines
```

Comment:
What's good for SEO is basically what's good for a human reader too, I wonder if shouldn't subtly reorient this towards "Best practices for structuring your pages". We should all be referring to this and absorbing it into our muscle memory over time!

I think this page is one of the most useful and immediately practical for all docs contributors, so I wonder if it might be buried and lost a bit in the style guide section.

To my eyes, ideally it would be a sibling to how to create good issues and how to write cumulative docs. 🤔 

But maybe that's something we can think about in a follow-up.